docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,412 @@ title: Data Grid - Editing
 
-{{"demo": "FullFeaturedCrudGrid.js", "bg": "inline", "disableAd": true}}
+```tsx
+<DataGrid editMode="row" columns={[{ field: 'name', editable: true }]} />
+```
 
-### Saving rows with nested structures
+The following demo illustrates how row editing works.
+The user can [start](#start-editing) and [stop](#stop-editing) editing a row using the same actions as those provided for cell editing (e.g. double-clicking a cell).
 
-Saving columns that make use of `valueGetter` can be done adding a `valueSetter`.
-The same [approach](/components/data-grid/editing/#saving-nested-structures) from the cell editing mode can be used here.
-Note that the `valueSetter` will be called for each field.
+{{"demo": "BasicRowEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Server-side persistence [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+> ⚠ By design, when changing the value of a cell all `preProcessEditCellProps` callbacks from other columns are also called.
+> This lets you apply conditional validation where the value of a cell impacts the validation status of another cell in the same row.
+> If you only want to run validation when the value has changed, check if the `hasChanged` param is `true`.
 
-If you want to send the updated data to your server, you can use the `onRowEditCommit` which is fired just before committing the new cell value to the grid.
+### Conditional validation
 
-To access the new values for the row, use `apiRef.current.getEditRowsModel` to get all rows in edit mode, then use the id provided to get only the values for the row that was committed.
+When all cells in a row are in edit mode, you can validate fields by comparing their values against one another.
+To do this, start by adding the `preProcessEditCellProps` as explained in the [validation](#validation) section.
+When the callback is called, it will have an additional `otherFieldsProps` param containing the props from the other fields in the same row.
+Use this param to check if the value from the current column is valid or not.
+Return the modified `props` containing the error as you would for cell editing.
+Once at the least one field has the `error` attribute set to a truthy value, the row will not exit edit mode.
 
-You can then decide if you want to send the whole row or only the modified fields, by checking them against the previous row values.
+The following demo requires a value for the **Payment method** column only if the **Is paid?** column is checked:
 
-{{"demo": "RowEditServerSidePersistence.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
+{{"demo": "ConditionalValidationGrid.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
 
-### Events [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+### Full-featured CRUD component [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
```

Comment:
I'm curious, is this a "Pro feature" only because it's using apiRef?

Reply:
Yes.